libwasmvm/src/api.rs

```diff
@@ -254,8 +262,46 @@ impl BackendApi for GoApi {
             Err(e) => return (Err(BackendError::unknown(e.to_string())), gas_info),
         };
         callee_instance.env.set_serialized_env(&contract_env);
-        callee_instance.set_storage_readonly(caller_env.is_storage_readonly());
         gas_info.cost += instantiate_cost;
+        // set read-write permission to callee instance
+        let callee_info = FunctionMetadata {
+            module_name: String::from(&func_info.module_name),
+            name: "_callee_func_list".to_string(),
+            signature: ([Type::I32], [Type::I32]).into(),
+        };
+        let serialized_param = to_vec(&String::from(&func_info.name)).unwrap();
+        let param_region_ptr = write_value_to_env(&callee_instance.env, &serialized_param).unwrap();
+        let callee_ret = match callee_instance.call_function_strict(
+            &callee_info.signature,
+            &callee_info.name,
+            &[param_region_ptr],
+        ) {
+            Ok(ret) => {
+                let ret_datas = match read_region_vals_from_env(
+                    &callee_instance.env,
+                    &ret,
+                    MAX_REGIONS_LENGTH_OUTPUT,
+                    true,
+                ) {
+                    Ok(v) => v,
+                    Err(e) => return (Err(BackendError::dynamic_link_err(e)), gas_info),
+                };
+                Ok(ret_datas)
+            }
+            Err(e) => Err(BackendError::dynamic_link_err(e.to_string())),
+        }
+        .unwrap();
+        let callee_func_map: HashMap<String, bool> =
+            serde_json::from_slice(&callee_ret[0]).unwrap();
+        let is_read_write_permission = callee_func_map[&func_info.name];
```

Comment:
It issues an error when the `&func_info.name` is not in the map. Catch the error and return as `Err(...)`.